### DISCUSSION TITLE
Clairfy data retention

### DISCUSSION THREAD

**Discussion by sandstrom** [lukevella/rallly#161] [Ideas]
I'm guessing old polls are cleared out after a month or two. Could be mentioned in the docs. Both for privacy and because it's good to know when the data will be wiped.

E.g. all data removed after ~3 months if there are no new replies. Or maybe 3 months after the last date in the poll, or after 12 months after creation, whichever comes first.

**Comment by lukevella** [comment]
At the moment there's nothing in place that would automatically clear old polls. The new version hasn't been live for long enough that any poll would require being deleted but I do of course plan on adding some endpoints to the API that will be used to clean up old polls, it's just not a major priority for the time being.

**Reply by sandstrom** [reply]
Alright, makes sense! Awesome thing you've built 🏅

**Comment by janru** [comment]
see https://support.rallly.co/#how-long-does-rallly-keep-a-poll-when-will-my-poll-be-deleted
The phrase "Update: As of 5 August 2022, polls will not be deleted unless all date options are in the past."
is not 100% clear to me.
Does it mean it will be deleted when all options are in the past **AND there was no edit for at least 30 days**?

Can you add an env variable to tune that?

**Reply by lukevella** [reply]
> Does it mean it will be deleted when all options are in the past AND there was no edit for at least 30 days?

Yes, both those conditions need to be met for a poll to be deleted. Can't think of a better way to phrase it right now but will try to revise this when I  do.

> Can you add an env variable to tune that?

To be clear this won't happen automatically for self-hosted instances because it relies on some external service to call the `/api/house-keeping` endpoint.

For the cloud-hosted instance, there is a [house-keeping workflow](https://github.com/lukevella/rallly/actions/workflows/house-keeping.yml) that runs once a day. All it does is hit the `/api/house-keeping` endpoint. I haven't really documented this but I would not have expected a self-hosted instance to run at a scale that would require this functionality but [the code is all there](https://github.com/lukevella/rallly/blob/main/src/pages/api/house-keeping.ts) and should be simple enough to understand.
